openedx/frontend-app-course-authoring
# [FC-0036] Refined tag drawer

## Description

Updates `openedx_learning` version 

## Supporting information

- Github issue: https://github.com/openedx/modular-learning/issues/190
- Related PRs:
    - https://github.com/openedx/edx-platform/pull/34490
    - https://github.com/openedx/openedx-learning/pull/180
- Internal ticket: [FAL-3645](https://tasks.opencraft.com/browse/FAL-3645)

## Testing instructions

- Use this branch of edx-platform: https://github.com/openedx/edx-platform/pull/34490
- Run `make requirements` on cms shell.
- Make sure you have some sample taxonomy/tags data setup: https://github.com/open-craft/taxonomy-sample-data
- Go into a course and click on "Manage tags" option for any unit/block
- Confirm:
    - Taxonomies with tags added to them are open by default, and ordered according to the number of tags they contain
    - Taxonomies with no tags are listed alphabetically below the taxonomies with tags
    - A footer with actions has been added to the drawer. In read-only mode it displays "Close" and "Edit Tags".
    - The new style of tags tree.
    - You are on read only mode: You can't add/remove tags.
    - You can close the drawer with the "Close" button
- Click on "Edit tags" and confirm:
    - X's appear next to deletable tags
    - Any collapsed taxonomies expand
    - "Add a tag" fields (https://github.com/openedx/modular-learning/issues/188) appear at the bottom of the sections for each taxonomy
    - Buttons in the footer change to "cancel" and "save"
    - Make sure the drawer scrolls
- Add and delete tags for any taxonomy
- Click on Cancel and verify that the drawer changes to read mode and all your changes are canceled.
- Click on "Edit tags" and add/delete tags and click "Save" and verify.
    - Your changes are saved.
    - The drawer changes to Read-Only mode but stays open
    - A toast appears in the bottom left of the page saying "N tags added", "N tags removed", or "X tags added, Y tags removed"
- Test the different states of the toast:
    - Add and remove tags.
    - Only add tags.
    - Only remove tags.
